### ISSUE TITLE
Ubuntu: Greyhole not starting

### ISSUE THREAD

**Issue by ralle12345** [gboudreau/Greyhole#152]
Hi,

after samba was updated this morning on my ubuntu 14.04 server greyhole won't start anymore.  THe current version of samba is 4.3.8+dfsg-0ubuntu0.14.04.2.

When greyhole is started, the following error is logged:

 ```
Apr 19 10:09:11 WARN initialize: Found a share (Movies) defined in /etc/greyhole.conf with no path in /etc/samba/smb.conf. Either add this share in /etc/samba/smb.conf, or remove it from /etc/greyhole.conf, then restart Greyhole.
Apr 19 10:09:11 CRITICAL initialize: Config file parsing failed. Exiting.
```

However, greping the smb.conf shows that there is a share with name 'Movies':

```
grep -A 5 Movies /etc/samba/smb.conf
[Movies]
        path = /storage/landingzone/movies
        read only = No
        create mask = 0775
        force create mode = 0775
        directory mask = 0775
```

I tried reinstalling the greyhole deb since samba was updated from 4.1.something to 4.3, but it didn't help to get up and running again. Is there any way to further increase the loglevel from DEBUG to INSANELYCHATTY to find out where the problem lies?

Thanks for any help you can provide!

**Comment by gboudreau** [comment]
Try: `testparm -s /etc/samba/smb.conf`
The output of that command is what Greyhole is using to know what is included in smb.conf

**Comment by gboudreau** [comment]
BTW, you will probably have issues trying to mount your Greyhole shares too.
Apparently, the samba-libs package on Ubuntu, for Samba 4.3, doesn't include the libraries that the Greyhole VFS module requires. What version of Ubuntu do you have installed?

Someone said he was able to manually compile the VFS using those instructions, and it worked:
https://github.com/gboudreau/Greyhole/blob/master/INSTALL#L85
Just make sure you use the exact Samba version you have installed.

**Comment by ralle12345** [comment]
Thanks for your help. I got it all up and running again. The Issue with greyhole not starting was caused by me having security=share in my smb.conf which led to a problem loading the config by testparm. 

The instructions on rebuilding the .so worked like a charm. At first I tried to take the easy route by trying to apt-get source samba, but that didn't work. After following you instructions it worked out. I didn't use the vfs_greyhole-samba-4.4.c source, but the vfs_greyhole-samba-4.3.c since I use samba 4.3, though.

Thanks for your extremely quick help!
